### PR TITLE
refactor(behavior_velocity_planner): use motion utils functions in crosswalk module

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Description

- use motion utils functions in crosswalk module

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_velocity_planner/src/scene_module/crosswalk/scene_walkway.cpp [hunk 1]
```diff
@@ -37,72 +43,127 @@ WalkwayModule::WalkwayModule(
...
+  return {};
+}
+
+bool WalkwayModule::modifyPathVelocity(PathWithLaneId * path, StopReason * stop_reason)
+{
+  const auto & base_link2front = planner_data_->vehicle_info_.max_longitudinal_offset_m;
+
   debug_data_ = DebugData();
-  debug_data_.base_link2front = planner_data_->vehicle_info_.max_longitudinal_offset_m;
-  first_stop_path_point_index_ = static_cast<int>(path->points.size()) - 1;
```
Comment (taikitanaka3) [inline planning/behavior_velocity_planner/src/scene_module/crosswalk/scene_walkway.cpp:45]: @satoshi-ota 
This is refactoring PR but do you want to remove this line?

Comment (satoshi-ota) [reply planning/behavior_velocity_planner/src/scene_module/crosswalk/scene_walkway.cpp:45]: I use motion_utils::insertTargetPoint() in walkway module and crosswalk module, so `first_stop_path_point_index_` is unused variable

**Review by taikitanaka3 — COMMENTED** [review]
@satoshi-ota 
for crosswalk LGTM
image
image

Walkway does not seems to be working, while it's registered.Can you check this?
image
```

[component_container_mt-13] [INFO] [1658890649.189800598] [planning.scenario_planning.lane_driving.behavior_planning.behavior_velocity_planner]: register task: module = walkway, id = 9284
[component_container_mt-13] [INFO] [1658890649.191015915] [planning.scenario_planning.lane_driving.behavior_planning.behavior_velocity_planner]: register task: module = intersection, id = 9205
[component_container_mt-13] [WARN] [1658890649.191337866] [planning.scenario_planning.lane_driving.behavior_planning.behavior_velocity_planner.RTCInterface[intersection]]: [isActivated] uuid : e10f2a07498b29a96e54cce9e9bf78d6 is not found.
[component_container_mt-13] 
[component_container_mt-13] [INFO] [1658890649.193890651] [planning.scenario_planning.lane_driving.behavior_planning.behavior_velocity_planner]: register task: module = blind_spot, id = 9205
[component_container_mt-13] [WARN] [1658890649.194187899] [planning.scenario_planning.lane_driving.behavior_planning.behavior_velocity_planner.RTCInterface[blind_spot]]: [isActivated] uuid : 41de221d753f255e480815ac4aa29a23 is not found.
```

**Review by taikitanaka3 — APPROVED** [review]
@satoshi-ota 
I found walkway virtual walll was missing at [autoware_launch](https://github.com/autowarefoundation/autoware_launch/blob/3fb59c335ed8a6de176d5d440f380f851ee78896/autoware_launch/rviz/autoware.rviz#L1054)

I add marker and it behaves as expected LGTM
image

**Comment by satoshi-ota** [comment]
@taikitanaka3 Thanks for your review :+1: I confirmed that there is no critical degradation in this PR by evaluator.
I'll merge this :smile:

**Pull request merged**